plugins-api/src/main/scala/za/co/absa/enceladus/plugins/api/Plugin.scala

```diff
@@ -18,4 +18,9 @@ package za.co.absa.enceladus.plugins.api
 /**
  * Base [marker] interface for all Enceladus plugins.
  */
-trait Plugin
+trait Plugin {
+  /**
+   * This method will be called when a plugin is no longer needed.
+   */
+  def close(): Unit
```

Comment:
If having a `close` method and this is a `trait` to be interface for outer development wouldn't an `init` method be also appropriate?

Reply:
Initialization is expected to be done in the constructor. But if later `init()` is required for some reason, we can always add it later. We can have a default implementation of doing nothing to not break existing plugins.

Reply:
Yeah, I realized after posting - the factory is good enough. Sorry.